eng/targets/Imports.targets

```diff
@@ -18,6 +18,12 @@
       which has all of our global settings 
     -->
     <UserRuntimeConfig Condition="'$(TargetFrameworkIdentifier)' == '.NETCoreApp' AND '$(OutputType)' == 'Exe'">$(RepositoryEngineeringDir)config\runtimeconfig.template.json</UserRuntimeConfig>
+
+    <DisableNullableWarnings Condition="'$(DisableNullableWarnings)' == '' AND '$(TargetFramework)' != '' AND '$(TargetFramework)' == 'netcoreapp3.1'">true</DisableNullableWarnings>
```

Comment:
Why is the second clause in the AND necessary? It would seem if the third clause (it's netcoreapp3.1) is true, the second is implied.

My question here is of course rooted in a deep mistrust of MSBuild and TFMs, so it wouldn't surprise me in the slightest if you tell me about something evil going on. But maybe that needs a comment then.

Reply:
> Why is the second clause in the AND necessary? It would seem if the third clause (it's netcoreapp3.1) is true, the second is implied.

In a multi-targetted build this file gets evaluated three times: once in the outer build where only `$(TargetFrameworks)` is defined and twice for each `$(TargetFramework)` value. Need the middle clause so we don't set the property on the outer evaluation.

Welcome to one hour of my life that I lost here ...

Reply:
> `'$(TargetFramework)' != '' AND '$(TargetFramework)' == 'netcoreapp3.1'`

This is an `A && B` scenario, where `B` implies `A`. It's possible the condition isn't written the expected way, because as-is the `'$(TargetFramework)' != ''` is unnecessary.

Reply:
Blarg I checked in a hack I was validiting locally with. Just a sec.

Reply:
The new warning check is the correct (albeit strange at a glance) one